client/src/tests/relatedItems/productCardsCarousel.test.jsx

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable no-undef */
+/* eslint-disable import/no-extraneous-dependencies */
+import React from 'react';
+import '@testing-library/jest-dom';
+import {
+  render, screen, cleanup, act, fireEvent,
+} from '@testing-library/react/pure';
+import userEvent from '@testing-library/user-event';
+// eslint-disable-next-line import/no-unresolved
+import Carousel from '../../components/relatedItemsAndComparison/Carousel.jsx';
+
+let carousel = null;
+
+const items = [
+  40348,
+  40352,
+  40350,
+];
+
+const wait = (t) => new Promise((r) => { setTimeout(r, t); });
+const reset = () => {
+  carousel = null;
+  cleanup();
+};
+
+let user;
+
+beforeEach(async () => {
+  user = userEvent.setup();
+  if (carousel === null) {
+    carousel = render(<Carousel items={items} classname="product-card" />);
+    await wait(3000);
+  }
+});
+
+describe('Product cards carousel renders 3 product cards', () => {
+  it('should render a slider with product cards', async () => {
+    expect(screen.getAllByTestId('slider')).not.toBe(null);
+    expect(screen.getAllByTestId('slider-product-card')).toHaveLength(items.length);
+  });
+
+  it('should render next slider button when the related products load and there are more than 1 related products', async () => {
+    expect(screen.findAllByTestId('product-card-next-button')).not.toBe(null);
+  });
+
+  it('should not render prev slider button when the related products load for the first time', async () => {
+    expect(screen.queryByTestId('product-card-prev-button')).not.toBeInTheDocument();
+  });
+
+  it('should render prev slider button when next button was clicked at least once', async () => {
+    act(() => {
+      // eslint-disable-next-line func-names
+      window.HTMLElement.prototype.scrollIntoView = function () {};
```

Comment:
You went a lot more into depth with testing than I did! I'm curious what the point of this line is.

Reply:
jsdom doesn't know what scrollIntoView is, that was a way to bring it in. I learned about it from here: https://stackoverflow.com/questions/53271193/typeerror-scrollintoview-is-not-a-function